miden-lib/asm/kernels/transaction/lib/memory.masm

```diff
@@ -1332,3 +1452,144 @@ end
 export.get_kernel_procedures_ptr
     push.KERNEL_PROCEDURES_PTR
 end
+
+# ACCOUNT DATA LOADERS
+# =================================================================================================
+
+#! Saves storage slots data into memory and validates that the storage commitment matches the 
+#! sequential storage hash.
+#!
+#! Inputs:
+#!   Operand stack: [STORAGE_COMMITMENT]
+#!   Advice map: {
+#!     STORAGE_COMMITMENT: [num_storage_slots, [STORAGE_SLOT_DATA]],
+#!   }
+#! Outputs:
+#!   Operand stack: []
+#! 
+#! Where:
+#! - STORAGE_COMMITMENT is the commitment of the foreign account's storage.
+#! - num_storage_slots is the number of foreign account's storage slots.
+#! - STORAGE_SLOT_DATA is the data contained in the storage slot which is constructed as follows: 
+#!   [SLOT_VALUE, slot_type, 0, 0, 0]
+#!
+#! Panics if:
+#! - the number of account storage slots exceeded the maximum limit of 255.
+#! - the computed account storage commitment does not match the provided account storage commitment
+export.save_account_storage_data
```

Comment:
I wonder if this (and the other) procedure should go into a different module. We usually try to keep `memory.masm` pretty light in terms of logic. So, maybe putting this into `accounts` module would be better? (unless it complicates something)